src/Symfony/Component/HttpKernel/HttpClientKernel.php

```diff
@@ -35,7 +35,7 @@ final class HttpClientKernel implements HttpKernelInterface
 
     public function __construct(HttpClientInterface $client = null)
     {
-        if (!class_exists(HttpClient::class)) {
+        if (null === $client && !class_exists(HttpClient::class)) {
```

Comment:
This should be reverted, right?

Reply:
Don't think so. There isn't any reason to force an instance of the `HttpClient` when the constructor allows any implementation of the `HttpClientInterface`.

Same check exists here, btw:
https://github.com/symfony/symfony/blob/8d450130e0422a8e056a0667dd730300f6be7a73/src/Symfony/Component/BrowserKit/HttpBrowser.php#L33-L35

Reply:
Understood, thanks.